python/pyspark/serializers.py

```diff
@@ -298,7 +298,10 @@ def create_array(s, t):
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             try:
-                array = pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)
+                if str(s.dtype) == 'category':
+                    array = pa.array(s.get_values())
```

Comment:
Instead of calling `s.get_values()` I think using `astype` with the category dtype is better. e.g.
```
s = s.astype(s.dtype.categories.dtype)
```

Reply:
Done